File: ProvidersService/src/main/java/com/platform/booking/recording/ProvidersService/models/WorkingHours.java
package com.platform.booking.recording.ProvidersService.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalTime;
import java.util.Objects;
import java.util.UUID;

@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "working_hours")

public class WorkingHours {
    @Id
    @Column(name = "id")
    @UuidGenerator(style = UuidGenerator.Style.TIME)
    private UUID id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "provider_id")
    private Provider provider;
    @Column(name = "day_of_week")
    private Integer dayOfWeek;
    @Column(name = "start_time")
    private LocalTime startTime;
    @Column(name = "end_time")
    private LocalTime endTime;
    @Column(name = "break_start_time")
    private LocalTime breakStartTime;
    @Column(name = "break_end_time")
    private LocalTime breakEndTime;
    @Column(name = "slot_step")
    private Integer slotStep;
    @Column(name = "is_active")
    private Boolean isActive;

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        WorkingHours that = (WorkingHours) o;
        return Objects.equals(id, that.id) && Objects.equals(dayOfWeek, that.dayOfWeek);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, dayOfWeek);
    }
}

File: FrontendService/src/app/core/models/working-hours.ts
export interface WorkingHoursGetDTO {
  id?: string;
  dayOfWeek: number;
  startTime?: string;
  endTime?: string;
  breakStartTime?: string;
  breakEndTime?: string;
  slotStep?: number;
  isActive: boolean;
}

export interface ListWorkingHoursGetDTO {
  workingHoursGetDTODTOList: WorkingHoursGetDTO[];
  providerId: string;
}

export interface WorkingHoursCreateDTO {
  dayOfWeek: number;
  startTime?: string;
  endTime?: string;
  breakStartTime?: string;
  breakEndTime?: string;
  slotStep?: number;
  isActive: boolean;
}

export interface ListWorkingHoursCreateDTO {
  workingHoursCreateDTOList: WorkingHoursCreateDTO[];
  providerId: string;
}
